go/libraries/doltcore/table/editor/creation/external_build_index.go

```diff
@@ -103,40 +104,50 @@ func BuildProllyIndexExternal(
 	}
 	defer sortedKeys.Close()
 
-	mut := secondary.Mutate()
 	it, err := sortedKeys.IterAll(ctx)
 	if err != nil {
 		return nil, err
 	}
 	defer it.Close()
 
-	var lastKey val.Tuple
+	tupIter := &tupleIterWithCb{iter: it, prefixDesc: prefixDesc, uniqCb: uniqCb}
+	ret, err := prolly.MutateMapWithTupleIter(ctx, secondary, tupIter)
+	if err != nil {
+		return nil, err
+	}
+
+	return durable.IndexFromProllyMap(ret), nil
+}
+
+type tupleIterWithCb struct {
+	iter sort.KeyIter
+	err  error
+
+	prefixDesc val.TupleDesc
+	uniqCb     DupEntryCb
+	lastKey    val.Tuple
+}
+
+var _ prolly.TupleIter = (*tupleIterWithCb)(nil)
+
+func (t *tupleIterWithCb) Next(ctx context.Context) (val.Tuple, val.Tuple) {
 	for {
-		key, err := it.Next(ctx)
+		curKey, err := t.iter.Next(ctx)
 		if err != nil {
-			if errors.Is(err, io.EOF) {
-				break
+			if !errors.Is(err, io.EOF) {
+				t.err = err
 			}
-			return nil, err
+			return nil, nil
```

Comment:
Why isn't the caller checking `tupIter.err` if this is our error reporting mechanism here?

Reply:
tru